lib/active_merchant/billing/gateways/komoju.rb

```diff
@@ -0,0 +1,84 @@
+require 'json'
+require 'komoju'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class KomojuGateway < Gateway
+      self.live_url = "https://gateway-sandbox.degica.com/api/v1"
+      self.supported_countries = ['JP']
+      self.default_currency = 'JPY'
+      self.money_format = :cents
+      self.homepage_url = 'https://komoju.com/'
+      self.display_name = 'Komoju'
+      self.supported_cardtypes = [:visa, :master, :american_express, :jcb]
+
+      def initialize(options = {})
+        requires!(options, :login)
+        @api_key = options[:login]
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        params = {
+          amount: amount(money),
+          description: options[:description],
+          payment_details: payment_details(payment, options),
+          currency: options[:currency] || currency(money),
+          external_order_num: options[:order_id], # should be removed when the param becomes optional
+          metadata: {
+            order_id: options[:order_id]
+          }
+        }
+        commit(params)
+      end
+
+      private
+
+      def payment_details(payment, options)
+        details = if payment.is_a?(CreditCard)
```

Comment:
This is an interesting way to get around the default "payment = credit card" assumption in activemerchant. FRT used different gateways for different payment types (`Gmo`, `GmoKonbini`) but since Komoju unifies these into a single `payment_details` hash it becomes much more sensible to treat them together.

I looked around and only found one other case of an activemerchant gateway doing multiple payment types: https://github.com/komoju/active_merchant/blob/c7b8b1a071807026e3ef0f24202b7d87fa3b9139/lib/active_merchant/billing/gateways/evo_ca.rb#L97

In this case it's [`Check`](https://github.com/komoju/active_merchant/blob/c7b8b1a071807026e3ef0f24202b7d87fa3b9139/lib/active_merchant/billing/check.rb#L8) and [`CreditCard`](https://github.com/komoju/active_merchant/blob/c7b8b1a071807026e3ef0f24202b7d87fa3b9139/lib/active_merchant/billing/credit_card.rb#L49), both defined in master. I wonder if it would make any sense to *add* other payment methods as models, e.g. `Konbini`, `BankTransfer`, etc. Then you could avoid the need for the `payment_details` hash which is not a standard option in activemerchant.